lib/operations/collection/get.js

```diff
@@ -1,46 +1,115 @@
 'use strict';
 
-function createList(map, context) {
-	const keys = Object.keys(map);
-	const query = context.query;
-	const result = [];
+const PAGE_AFTER = 'pageAfter';
+const PAGE_NUM = 'pageNum';
+const PAGE_SIZE = 'pageSize';
 
-	for (let i = 0; i < keys.length; i++) {
-		const id = keys[i];
-		const resource = map[id];
+function getProperties(query) {
+	const properties = {
+		matchOn: {}
+	};
+	for (const key in query) {
+		if (key === PAGE_SIZE) {
+			properties.pageSize = parseInt(query[PAGE_SIZE], 10);
+		} else if (key === PAGE_NUM) {
+			properties.pageNum = parseInt(query[PAGE_NUM], 10);
+		} else if (key === PAGE_AFTER) {
+			properties.pageAfter = query[PAGE_AFTER];
+		} else {
+			properties.matchOn[key] = query[key];
+		}
+	}
+	return properties;
+}
+
+function validateProperties(map, properties) {
+	let status = 200;
+
+	const hasPageSize = properties.hasOwnProperty('pageSize');
+	const pageSizeIsNumber = Number.isInteger(properties.pageSize) && typeof (properties.pageSize) === 'number';
```

Comment:
Another bit I missed by being too shortsighted with my changes. I'll have to pay more attention to full lines in future.